### DISCUSSION TITLE
Problem with running AllChem.EmbedMultipleConfs: Failed Expression: ub >= lb

### DISCUSSION THREAD

**Discussion by lucyraven** [rdkit/rdkit#6617] [Q&A]
Hi!
I'm generating 3D conformations of a series of macrocyclic compounds, but for some of the compounds I get the following error before my script crashes (the error is not consistent and the same molecule that sometimes crashes the script will other times work fine):

"Invariant Violation

Violation occurred on line 55 in file /project/build/temp.linux-x86_64-cpython-311/rdkit/Code/DistGeom/DistGeomUtils.cpp
Failed Expression: ub >= lb
****
...
Violation occurred on line 55 in file Code/DistGeom/DistGeomUtils.cpp
	Failed Expression: ub >= lb
	RDKIT: 2023.03.2
	BOOST: 1_78"

The code I'm running:
    mol = Chem.MolFromSmiles(smiles)
    mol = Chem.AddHs(mol)

    params = AllChem.ETKDGv3() #contains the modified macrocycle torsion workflow
    bmat = cpeptools.bound_matrix_from_ellipse(mol, angle = 0, eccentricity = 0.99)
    params.SetBoundsMat(bmat)
    amide_cpci = cpeptools.get_amide_pairwise_coulomb_interaction(mol)
    params.SetCPCI(amide_cpci)
    params.numThreads=0
    params.pruneRmsThresh=0.3
    params.enforceChirality=True

    AllChem.EmbedMultipleConfs(mol,sampling,params)

I went to the code and can see that this error is thrown from the function pickRandomDistMat within DistGeomUtils.cpp, but I don't know what it is about the molecules that poses a problem and how to circumvent it. I checked the SMILES string I started from and it is readable and produces a "normal" macrocycle.

Does anyone have some helpful hints or ideas for troubleshooting this problem? 

Kindly,
Lucy

**Comment by greglandrum** [comment]
Hi @lucyraven ,
This error happens when an upper bound in the distance bounds matrix is lower than the corresponding lower bound.

I'm guessing that cpeptools is making some kind of mistake when it sets up the bounds matrix. Can you share the SMILES that is causing the problem?

**Reply by lucyraven** [reply]
Hi Greg, 
Unfortunately I cannot share the SMILES as I'm bound by an NDA. A common denominater for the 7 compounds that fail is a large macrocycle ring with at least 3 traditional ring structures as part of the ring. Is it pointless to troubleshoot without the actual SMILES? And should I maybe talk to the cpeptools guys directly?

Thanks,
Lucy

**Reply by lucyraven** [reply]
I now finally have a non-NDA-bound SMILES string we can troubleshoot from: S2c1ccc(cc1)OCCC=CCC2

Generally, it seems the problem occurs for macrocyclic rings that contain a ring. The issue seems to be that DistanceGeometry.DoTriangleSmoothing fails and messes up the bmat matrix while failing.

In case it is a cpeptools problem, I have raised an issue with them too.
